prefect_azure/blob_storage.py

```diff
@@ -177,3 +189,472 @@ def example_blob_storage_list_flow():
+            for reading and writing objects.
+    """
+
+    _block_type_name = "Azure Blob Storage Container"
+    _logo_url = "https://cdn.sanity.io/images/3ugk85nk/production/54e3fa7e00197a4fbd1d82ed62494cb58d08c96a-250x250.png"  # noqa
+    _documentation_url = "https://prefecthq.github.io/prefect-azure/blob_storage/#prefect_azure.blob_storabe.AzureBlobStorageContainer"  # noqa
+
+    container_name: str = Field(
+        default=..., description="The name of a Azure Blob Storage container."
+    )
+    credentials: AzureBlobStorageCredentials = Field(
+        default_factory=AzureBlobStorageCredentials,
+        description="The credentials to use for authentication with Azure.",
+    )
+    base_folder: Optional[str] = Field(
+        default=None,
+        description=(
+            "A base path to a folder within the container to use "
+            "for reading and writing objects."
+        ),
+    )
+
+    def _get_path_relative_to_base_folder(self, path: Optional[str] = None) -> str:
+        if path is None and self.base_folder is None:
+            return ""
+        if path is None:
+            return self.base_folder
+        if self.base_folder is None:
+            return path
+        return (Path(self.base_folder) / Path(path)).as_posix()
+
+    @sync_compatible
+    async def download_folder_to_path(
+        self,
+        from_folder: str,
+        to_folder: Union[str, Path],
+        **download_kwargs: Dict[str, Any],
+    ) -> Coroutine[Any, Any, Path]:
+        """Download a folder from the container to a local path.
+
+        Args:
+            from_folder: The folder path in the container to download.
+            to_folder: The local path to download the folder to.
+            **download_kwargs: Additional keyword arguments passed into
+                `BlobClient.download_blob`.
+
+        Returns:
+            The local path where the folder was downloaded.
+
+        Example:
+            Download the contents of container folder `folder` from the container
+                to the local folder `local_folder`:
+
+            ```python
+            from prefect_azure import AzureBlobStorageCredentials
+            from prefect_azure.blob_storage import AzureBlobStorageContainer
+
+            credentials = AzureBlobStorageCredentials(
+                connection_string="connection_string",
+            )
+            block = AzureBlobStorageContainer(
+                container_name="container",
+                credentials=credentials,
+            )
+            block.download_folder_to_path(
+                from_folder="folder",
+                to_folder="local_folder"
+            )
+            ```
+        """
+        self.logger.info(
+            "Downloading folder from container %s to path %s",
+            self.container_name,
+            to_folder,
+        )
+        full_container_path = self._get_path_relative_to_base_folder(from_folder)
```

Comment:
I think it would be good to verify that the from_folder exists by asserting that the blob is not empty or something or blob.name.

Reply:
That would be good, but it is surprisingly difficult to do. Can you create an issue to add that to another PR?

Reply:
👍